packages/ai-cloudflare/src/index.ts

```diff
@@ -0,0 +1,38 @@
+import type { Connection, ConnectionContext } from "agents";
+import { AIChatAgent } from "agents/ai-chat-agent";
+
+export const REFRESH_HEADER = "x-refresh-token";
+
+export abstract class FederatedConnectionChatAgent<
+  Env = unknown,
+  State = unknown,
+> extends AIChatAgent<Env, State> {
+  refreshToken: string = "";
+
+  async onRequest(request: Request) {
+    request = this.#extractRefreshToken(request);
+
+    const { pathname, searchParams } = new URL(request.url);
+    if (pathname.endsWith("/refresh")) {
```

Comment:
If I understand correctly the goal of this code path is to update the `refreshToken` once the popup is closed, is it right?